### DISCUSSION TITLE
Fichier arrêté à télécharger

### DISCUSSION THREAD

**Discussion by OUARZA** [OUARZA/Propluvia#34] [Ideas]
#Le numéro du fichier se cache dans l’API sous le code fdCdn

![image](https://github.com/OUARZA/Propluvia/assets/34892335/7da481f7-7450-416f-abe0-7a5e8c9448b7)

Pour le télécharger, le lien est https://eau.api.agriculture.gouv.fr/apis/propluvia/file/pdf/{code fdCdn}
Comme ici [https://eau.api.agriculture.gouv.fr/apis/propluvia/file/pdf/223397 1](https://eau.api.agriculture.gouv.fr/apis/propluvia/file/pdf/223397)

https://github.com/users/OUARZA/projects/2?pane=issue&itemId=35129901

**Comment by elmerdu69** [comment]
voilà c'est à tester sur la prochaine béta

**Reply by OUARZA** [reply]
J'ai pensé à un truc.
Avoir une commande non affichée avec le lien brut (pour pouvoir l'utiliser dans un scénario sms par exemple), et afficher sur le plugin au format `"<a href="##COMMANDE##">Arrêté n°##NUMEROARRETE##</a>"`

**Reply by elmerdu69** [reply]
oui c'est tout a fait possible, je ne voulais pas surcharger d'une commande supplémentaire. Si s'est utile je ferai la modif.

**Reply by OUARZA** [reply]
Je me suis fais un scenario, du coup si je veux recevoir le lien cliquable par SMS (ou même autre), je pense qu'il faut la valeur brute.

**Reply by elmerdu69** [reply]
PR poussé sur la beta : l'url complète est maintenant stocké dans la commande "Url arrêté en pdf"
Cette commande sera créée en cliquant sur le bouton sauvegarder sur la page de l'équipement

**Comment by OUARZA** [comment]
3 remarques:
- quand il n'y a pas d'arrêté, y'a un lien "fictif"
- la couleur en mode sombre est pas top
- possible de mettre un _blank pour ouvrir le lien dans un autre onglet sans écraser l'onglet actif ?
- 
![image](https://github.com/OUARZA/Propluvia/assets/34892335/1e68064d-92b8-4bb6-9724-bfd988db90d7)

**Reply by elmerdu69** [reply]
PR poussé sur la beta : 
- pas de lien généré si pas d'arrêté
- ouverture du lien dans un nouvel onglet (via target = "_blank" -> Firefox télécharge le pdf et l'ouvre dans un nouvel onglet, Edge n'affiche pas de page mais télécharge le PDF, chrome pas testé
- ajout  d'une classe CSS pour modifier la couleur d'affichage. Malheureusement j'ai l'impression que la classe CSS du core  jeedom à la priorité sur celle que j'ai créée. Même en ajoutant un style dans la balise, ça ne marche pas. Regarde de ton coté, peut être tu arriveras à comprendre mieux que moi

**Reply by OUARZA** [reply]
Hello,
je pense avoir réussi à appliquer du noir sur le lien avec : 
.a:link{
 color: black !important;
}
Mais il faut voir comment c'est géré en thème sombre/clair.
je n'ai pas trouvé pour modifier la couleur lors du survole du lien...
je te laisse regarder, je n'ai pas modifié le CSS.

De mon côté le lien ne marche plus :(
![image](https://github.com/OUARZA/Propluvia/assets/34892335/4a4c1515-8818-4737-9a90-1904a29f4644)

**Reply by elmerdu69** [reply]
Hello le lien ne marche plus suite à ma modif d'hier? De mon coté ça marche sans problème.

**Reply by OUARZA** [reply]
Oui

**Reply by elmerdu69** [reply]
étrange.. tu es sous quel navigateur ? tu peux en tester un autre?

**Reply by elmerdu69** [reply]
je teste ta commune, je tombe sur un timeout...

**Reply by OUARZA** [reply]
Problème réglé...
Il fallait que je resauvegarde l'équipement, actualise les données et c'est bon.
Je te confirme que le pdf se télécharge bien sous Edge. Je teste ce soir sur chrome.

**Reply by elmerdu69** [reply]
ah ah j'avais bien précisé dans le commentaire d'en haut qu'il fallait d’abord recliquer sur le bouton sauvegarder ;)
Tous vas bien du coup mais tu m'as fais une bonne frayeur pendant 10 min

Tu as le timeout en ce moment chez toi?

**Reply by elmerdu69** [reply]
tu l'as appliqué à quel endroit le 
```
.a:link{
color: black !important;
}
```
?

**Reply by OUARZA** [reply]
Oui, tu as raison, j'en suis désolé.
Non pas de timeout.

Dans le fichier propluvia.dashboard.css

**Reply by elmerdu69** [reply]
ça y est j'ai quelque chose d'à-peu-près acceptable en jouant avec _!important_ :
light + lien normal
![image](https://github.com/OUARZA/Propluvia/assets/115868585/7cde9d96-4199-45bd-a003-111f0aca2bfe)
light + curseur souris sur le lien
![image](https://github.com/OUARZA/Propluvia/assets/115868585/e0539e49-1ef5-49ee-bba7-a88f73b0e242)
dark + lien normal
![image](https://github.com/OUARZA/Propluvia/assets/115868585/52f23933-296e-4cbf-a472-259352a7c479)
dark + curseur souris sur le lien
![image](https://github.com/OUARZA/Propluvia/assets/115868585/d61085cf-bd44-4ea9-b8d0-fe8050f5aaa9)

Donc maintenant je peux
 - mettre la couleur que je veux sur le lien normal (ici bleu mais on peut mettre du noir, du rouge ...)
 - mettre le soulignement ou pas du lien
 Je ne sais encore pas faire mais je cherche encore
- changer la couleur du lien quand la souris est dessus

Qu'en penses-tu ? que préfères-tu comme config sur le lien?

**Reply by elmerdu69** [reply]
...trop rapide dans mon message 
j'ai trouvé maintenant comment changer la couleur quand le curseur est sur le lien.

Maintenant tu n'as plus qu'a me dire dire ce que tu préfères. 
Perso j'aime bien le lien en bleu souligné (ou pas) et qui passe en rouge souligné quand la souris est dessus
![image](https://github.com/OUARZA/Propluvia/assets/115868585/2b06953f-73ad-478b-bee7-8c8314433333)

et merci pour le tuyau qui m'a mis sur la piste ;-)

**Reply by OUARZA** [reply]
Avec plaisir.
En fait, soit on choisit nos couleurs, soit on garde sur le standard du web.
Le bleu/rouge me va.

**Reply by elmerdu69** [reply]
va pour ça alors

**Reply by OUARZA** [reply]
Je passe le PR demain, j'en ai déjà passé un aujourd'hui.
Je trouve dommage le peu de retour, mais ça ne m'étonne pas.
